tests/integration/study_data_blueprint/test_table_mode.py

```diff
@@ -188,25 +189,176 @@ def test_lifecycle__nominal(
+                        "usePhaseShifter": True,
+                    },
+                    "fr / it": {
+                        "assetType": "DC",  # case-insensitive
+                    },
+                },
+            )
+            assert res.status_code == 200, res.json()
+
+            expected_links = {
                 "de / fr": {
-                    "colorRgb": "#FFA500",
+                    "area1": "de",
+                    "area2": "fr",
+                    "assetType": "ac",
+                    "colorb": 100,
+                    "colorg": 150,
+                    "colorr": 200,
+                    "displayComments": False,
+                    "hurdlesCost": True,
+                    "linkStyle": "plain",
+                    "linkWidth": 2,
+                    "loopFlow": False,
+                    "transmissionCapacities": "ignore",
+                    "usePhaseShifter": False,
+                },
+                "de / it": {
+                    "area1": "de",
+                    "area2": "it",
+                    "assetType": "ac",
+                    "colorr": 112,
+                    "colorg": 112,
+                    "colorb": 112,
+                    "displayComments": True,
+                    "hurdlesCost": False,
+                    "linkStyle": "plain",
+                    "linkWidth": 1,
+                    "loopFlow": False,
+                    "transmissionCapacities": "enabled",
+                    "usePhaseShifter": False,
+                },
+                "es / fr": {
+                    "area1": "es",
+                    "area2": "fr",
+                    "assetType": "ac",
+                    "colorb": 100,
+                    "colorg": 150,
+                    "colorr": 200,
+                    "displayComments": True,
+                    "hurdlesCost": True,
+                    "linkStyle": "plain",
+                    "linkWidth": 1,
+                    "loopFlow": False,
+                    "transmissionCapacities": "enabled",
+                    "usePhaseShifter": True,
+                },
+                "fr / it": {
+                    "area1": "fr",
+                    "area2": "it",
+                    "assetType": "dc",
+                    "colorb": 112,
+                    "colorg": 112,
+                    "colorr": 112,
+                    "displayComments": True,
+                    "hurdlesCost": True,
+                    "linkStyle": "plain",
+                    "linkWidth": 1,
+                    "loopFlow": False,
+                    "transmissionCapacities": "enabled",
+                    "usePhaseShifter": False,
+                },
+            }
+
+            # asserts actual equals expected without the non-updated link.
+            actual = res.json()
+            expected_result = copy.deepcopy(expected_links)
+            del expected_result["de / it"]
+            assert actual == expected_result
```

Comment:
This test file is already really long (1000+ lines), could you store the JSON inside a variable instead of copying it everywhere as it makes the test hard to read

Reply:
I extracted the JSON input and the expected links into another method that determines the desired output based on the version. While it is not shorter in terms of line count, it is definitely easier to read